packages/salesforcedx-vscode-core/src/commands/forceStartApexDebugLogging.ts

```diff
@@ -65,50 +78,69 @@ export class ForceStartApexDebugLoggingExecutor extends SfdxCommandletExecutor<{
           traceflag.DebugLevel.ApexCode,
           traceflag.DebugLevel.Visualforce
         );
-        developerLogTraceFlag.turnOnLogging();
 
-        const updateDebugLevelsCommandlet = new SfdxCommandlet(
-          new SfdxWorkspaceChecker(),
-          new EmptyParametersGatherer(),
-          new UpdateDebugLevelsExecutor()
-        );
-        const promises = [updateDebugLevelsCommandlet.run()];
+        execution = new CliCommandExecutor(
+          new UpdateDebugLevelsExecutor().build(),
+          {
+            cwd: vscode.workspace.rootPath
+          }
+        ).execute(cancellationToken);
+        this.attachSubExecution(execution);
+        resultPromise = new CommandOutput().getCmdResult(execution);
+        result = await resultPromise;
 
-        let updateTraceFlagCommandlet;
         if (!developerLogTraceFlag.isValidDateLength()) {
           developerLogTraceFlag.validateDates();
-          updateTraceFlagCommandlet = new SfdxCommandlet(
-            new SfdxWorkspaceChecker(),
-            new EmptyParametersGatherer(),
-            new UpdateTraceFlagsExecutor()
-          );
-        }
-        if (updateTraceFlagCommandlet) {
-          promises.push(updateTraceFlagCommandlet.run());
+          execution = new CliCommandExecutor(
+            new UpdateTraceFlagsExecutor().build(),
+            {
+              cwd: vscode.workspace.rootPath
+            }
+          ).execute(cancellationToken);
+          this.attachSubExecution(execution);
+          resultPromise = new CommandOutput().getCmdResult(execution);
+          result = await resultPromise;
+          resultJson = JSON.parse(result);
         }
-
-        const updatePromise = Promise.all(promises);
-        await updatePromise;
       } else {
-        const createDebuglevelCommandlet = new SfdxCommandlet(
-          new SfdxWorkspaceChecker(),
-          new EmptyParametersGatherer(),
-          new CreateDebugLevel()
+        execution = new CliCommandExecutor(new CreateDebugLevel().build(), {
+          cwd: vscode.workspace.rootPath
+        }).execute(cancellationToken);
+        this.attachSubExecution(execution);
+        resultPromise = new CommandOutput().getCmdResult(execution);
+        result = await resultPromise;
+        resultJson = JSON.parse(result);
+        const debugLevelId = resultJson.result.id;
+        developerLogTraceFlag.setDebugLevelInfo(
+          debugLevelId,
+          APEX_CODE_DEBUG_LEVEL,
+          VISUALFORCE_DEBUG_LEVEL
         );
-        await createDebuglevelCommandlet.run();
+
         const userId = await getUserId(
           vscode.workspace.workspaceFolders![0].uri.fsPath
```

Comment:
Is this the one that is safe to use on Windows? I had some issues with something on LWC the other day. Could you check what I used in the end and emulate that. It's in one of the PRs in this repo.

Reply:
Yes this is the one that's safe on Windows. You're using the same thing in PR #283